dashboard/app/controllers/javabuilder_connection_test_controller.rb

```diff
@@ -0,0 +1,23 @@
+require 'cdo/firehose'
+
+class JavabuilderConnectionTestController < ApplicationController
+  before_action :authenticate_user!
+
+  # POST /javabuilder/connectivity_test_logging
+  def log
+    data = {user_type: current_user.user_type}
+    if params[:error]
+      data[:error] = params[:error]
+    end
+
+    FirehoseClient.instance.put_record(
+      :analysis,
+      {
+        study: 'java-builder-connectivity',
```

Comment:
nit: should `javabuilder` be one word since don't typically break it up elsewhere?

Reply:
I think I saw this example and followed it:

https://github.com/code-dot-org/code-dot-org/blob/44d97b064a2050bfd678b1f1d258932fb61c7383/dashboard/app/controllers/javabuilder_sessions_controller.rb#L57

But yeah one word probably makes more sense.

Reply:
Ahh got it, probably better to be consistent then so these two are linked